### PR TITLE
GITC-102: Hackathons landing page (developers) refresh

### PR DESCRIPTION
**Pull request by gdixon**
##### Description

This PR implements the hackathons landing page refresh.

build design that includes:

- [x] hackathon landing page header & footer CTAs for exploring & becoming a sponsor
- [x] hackathons explorer preview (lead-in to CTA)
- [x] hackathon positive results from developers (testimonials)
- [x] hackathon landing page CTA to read instructions on how to get started (happyfox)
- [x] responsive views

##### Refers/Fixes

Fixes: GITC-102
Fixes: GITC-232

##### Testing

Tested locally

localhost_8000_hackathon-list

### CODE REVIEW

File: app/app/templates/shared/hackathon-list.html [hunk 1]
```diff
@@ -1,87 +1,287 @@
...
+                      <div class="mt-3 font-smaller-1 line-height-2 text-grey-400 hackathon-summary {% if left_align_cards %}text-left{% endif %}" style="min-height:80px;">
+                        <p>{{ event.summary }}</p>
+                      </div>
+
+                      <div class="mt-auto text-left">
+                        {% if event.sponsor_profiles|length %}
+                          <div class="mt-3 font-smaller-2">
+                            <div class="mb-1 text-grey-400">Sponsored by</div>
+                            {% for sponsor in event.sponsor_profiles|slice:":6" %}
+                              <a href="{{ sponsor.absolute_url }}" target="_blank"><img class="rounded-circle my-1 mr-1" style="width:36px;" src="{{ sponsor.avatar_url }}" /></a>
```
Comment (thelostone-mc) [inline app/app/templates/shared/hackathon-list.html:56]: worth adding `noopener/noreferrer` ?

Comment (gdixon) [reply app/app/templates/shared/hackathon-list.html:56]: We could do, but these are internal links - they're pointing to the sponsors profile (this code is unchanged from the original `hackathon-list.html` page)

**Review by zacheryschiller — APPROVED** [review]
Looks really good! I pulled it and tested locally too. I think it's good to move forward. We need to make sure the content as the HappyFox link is updated before we deploy (https://support.gitcoin.co/kb/article/81-hackathons-how-it-works)

**Comment by owocki** [comment]
some feedback. some is about design, some about implmenetation
- current hackathon landing page has a big giant 'contact sales' at top but this one doesnt.  why is it so buried below the fold?
- is the testimonials module meant to just have 1 sad lonely testimonial from cryptomental or is it meant to have many u can scroll thru? why doesnt cryptomental have a pic?
- i get a 500 when i click to any hackathon  https://stage.gitcoin.co/hackathon/sovrython/onboard

**Comment by gdixon** [comment]
> why doesnt cryptomental have a pic?

We're pulling that image from http://stage.gitcoin.co/dynamic/avatar/cryptomental - for some reason the dynamic end-point on staging is failing and I've not had chance to dig into why yet (its likely because we don't have s3 connected and all of this data was copied from production without any of the assets).

This works on both my local and on production (https://gitcoin.co/dynamic/avatar/cryptomental/) - I can have a look to get it going again tomorrow.

> * i get a 500 when i click to any hackathon  https://stage.gitcoin.co/hackathon/sovrython/onboard

Ahhh, sorry about that, I hadn't ran the migrations - that should work now -- please note that I've placed the JSONstore manually to reflect the state of production, so some hackathons won't be present on staging.

**Review by PixelantDesign — COMMENTED** [review]
Lookin' good! 

**Header**
I'm noticing extra space between
_Hey Hackers, go build cool stuff_ and _Build a bunch of cool stuff, learn new tech, support the open source ecosystem, and earn cash and prizes!_. Consider removing extra space. See mission page header spacing.

**Sponsor Hackathon**
_Want to become a sponsor or run an exclusive hackathon on Gitcoin?_
This feels like a system alert (less of a promotion), as it follows our system alert design.

**How to Get Started**
Could we consider switching the _How to get started_ section and the _Get the latest on Gitcoin email newsletter_ section? 

**Email Newsletter**
Email newsletter section, when I entered my email, I didn't receive a confirmation message (see mission page newsletter module).

cc @willsputra

**Comment by willsputra** [comment]
> current hackathon landing page has a big giant 'contact sales' at top but this one doesnt. why is it so buried below the fold?

>Sponsor Hackathon
Want to become a sponsor or run an exclusive hackathon on Gitcoin?
This feels like a system alert (less of a promotion), as it follows our system alert design.

Since this page is more optimised for hackers, we want explore hackathons to be the primary CTA. We also wanted to provide a bit more context for funders that they can either be a sponsor or run exclusive hacks with us.

We can update the design to make it more obvious tho (and not clash with our system design) as below cc @PixelantDesign 
@gdixon would it be difficult to change the sponsor section design to this?

Screenshot 2021-07-20 at 8 28 42 PM

> I'm noticing extra space between
Hey Hackers, go build cool stuff and Build a bunch of cool stuff, learn new tech, support the open source ecosystem, and earn cash and prizes!. Consider removing extra space. See mission page header spacing.

> Email Newsletter
Email newsletter section, when I entered my email, I didn't receive a confirmation message (see mission page newsletter module).

agree with these.

> How to Get Started
Could we consider switching the How to get started section and the Get the latest on Gitcoin email newsletter section?

I'd still prefer putting the How to get started at the bottom so new users understand the what > why > how

---

A couple more questions:
1. Are we building the new Prizes field? We talked about having a new field for prizes in USD that admins can enter manually.
Screenshot 2021-07-20 at 8 36 43 PM

2. For the floating bubbles, I understand we went with hackathon logos due to performance reasons. Would there be alternative ways to display org logos instead, say maybe pulling all org logos that are hackathon sponsors instead of pulling from leaderboard?
Because the purpose of the bubbles is to inform "hey here's a lot of big-name projects you can work with" (for hackers) and "hey here are other big-name projects that have run hackathons with us, so don't miss out" (for sponsors), it'd be better to have recognisable org logos. But if it the effort or performance hit is too big, this is fine for now 🙂

**Comment by gdixon** [comment]
> We can update the design to make it more obvious tho (and not clash with our system design)

This is resolved in b2e2acd

> I'm noticing extra space between
> > Hey Hackers, go build cool stuff and Build a bunch of cool stuff, learn new tech, support the open source ecosystem, and earn cash and prizes!. Consider removing extra space. See mission page header spacing.

This is resolved in 3f09665

> Email newsletter section, when I entered my email, I didn't receive a confirmation message (see mission page newsletter module).

Nice catch @PixelantDesign - thats now resolved in b2e2acd

> A couple more questions:
> 
> 1. Are we building the new Prizes field? We talked about having a new field for prizes in USD that admins can enter manually.

Yes, that will be picked up when it is filled in the backend, the data on staging was a duplicate of whats on production, and the field won't be available until this PR is deployed. 

> 2. For the floating bubbles, I understand we went with hackathon logos due to performance reasons. Would there be alternative ways to display org logos instead, say maybe pulling all org logos that are hackathon sponsors instead of pulling from leaderboard?

Yes, let's split this out into a separate ticket, we will have to have a think about how we curate that list to ensure all sponsors are highlighted fairly.

**Review by chibie — APPROVED** [review]
lgtm!

**Comment by gkimbwala** [comment]
This looks pretty good to me. There is a minor detail that catches my eye. I am not sure why the "join Now" buttons and  "prizes" label on the current hackathons are two different colors. A little confused by that but other than that it looks good.
Screen Shot 2021-07-22 at 9 48 02 AM

**Comment by gdixon** [comment]
> This looks pretty good to me. There is a minor detail that catches my eye. I am not sure why the "join Now" buttons and "prizes" label on the current hackathons are two different colors. A little confused by that but other than that it looks good.

This state shows when the hackathon event has reached an end but before the page cache has been regenerated (which would move the event in to the `completed` tab). 

In the old design those buttons were removed, but because that would leave a lot of empty space, I opted to disable them instead. If someone from the team was to go in to the _admin panel, they would be able to trigger the cache to be regenerated manually (or we can wait for the cron to trigger the refresh (it runs once every 3 hours)). 

So this state is rare, but it does handle an edge-case (you cannot join a hackathon after it has ended).

**Comment by gkimbwala** [comment]
> > This looks pretty good to me. There is a minor detail that catches my eye. I am not sure why the "join Now" buttons and "prizes" label on the current hackathons are two different colors. A little confused by that but other than that it looks good.
> 
> This state shows when the hackathon event has reached an end but before the page cache has been regenerated (which would move the event in to the `completed` tab).
> 
> In the old design those buttons were removed, but because that would leave a lot of empty space, I opted to disable them instead. If someone from the team was to go in to the _admin panel, they would be able to trigger the cache to be regenerated manually (or we can wait for the cron to trigger the refresh (it runs once every 3 hours)).
> 
> So this state is rare, but it does handle an edge-case (you cannot join a hackathon after it has ended).

Got it! Then it looks good to me.

File: app/retail/templates/about.html [hunk 2]
```diff
@@ -290,112 +290,12 @@ <h2>Get the latest from Gitcoin</h2>
     </div>
     {% include 'shared/footer.html' %}
     {% include 'shared/footer_scripts.html' with slim=1 %}
-    <script src="{% static "v2/js/lib/svgator_player.js" %}"></script>
     <script src="https://cdn.jsdelivr.net/npm/perlin@1.0.0/index.min.js"></script>
+    <script src="{% static "v2/js/lib/svgator_player.js" %}"></script>
+    <script src="{% static "v2/js/bubbles.js" %}"></script>
```
Comment (octavioamu) [inline app/retail/templates/about.html:295]: goood one

File: app/dashboard/views.py [hunk 3]
```diff
@@ -5415,29 +5416,53 @@ def hackathon_registration(request):
...
-    num_upcoming = len([ele for ele in events if ele['type'] == 'upcoming'])
-    num_finished = len([ele for ele in events if ele['type'] == 'finished'])
     tabs = [
         ('current', 'happening now', num_current),
         ('upcoming', 'upcoming', num_upcoming),
         ('finished', 'completed', num_finished),
     ]
 
+    # shuffle the sponsors
+    random.shuffle(sponsors)
```
Comment (octavioamu) [inline app/dashboard/views.py:5446]: 🔥

**Review by octavioamu — APPROVED** [review]

**Pull request merged**